### PR TITLE
[tbl111] Use pipenv for deterministic builds

### PR DESCRIPTION
**Pull request by andrewmil**
Pipenv enables us to lock down and guarantee what dependencies we get
when we install dependencies. This works by using the pip hashes when
running pipenv install --dev. The hashes are stored in a generated
file call Pipfile.lock.

We no longer need requirement files instead pipenv uses a file called
Pipfile. The documentation can be found
https://github.com/kennethreitz/pipenv.

https://trello.com/c/TNrTo9L1/733-tbl111-create-reproducible-deployments

How to review

Check out locally and follow README.md to set up with pipenv
cf push to check deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,20 +73,20 @@ Quick Start on localhost
...
+        pipenv run oauth2server/manage.py runserver 0.0.0.0:8040
+    ```
 
-- -  or the gunicorn web server
-```
-    gunicorn --bind 0.0.0.0:8040 --workers 4 proj.wsgi --pythonpath 'oauth2server'
-```
+    * or the gunicorn web server
+    ```
+        gunicorn --bind 0.0.0.0:8040 --workers 4 proj.wsgi --pythonpath 'oauth2server'
```
Comment (jcox-dev) [inline README.md]: Does `pipenv run` need to be appended here as well?

File: start.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
 
 ./init_db.sh
-python oauth2server/manage.py loaddata oauth2server/apps/credentials/fixtures/ons_credentials.json
-python oauth2server/manage.py runserver 0.0.0.0:8040
\ No newline at end of file
+pipenv run oauth2server/manage.py loaddata oauth2server/apps/credentials/fixtures/ons_credentials.json
```
Comment (jcox-dev) [inline start.sh]: I would leave these as `python...` as it's called from the Dockerfile (with system-installed deps).

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Looking consistent with the rest of the repos, which is great considering it's Py2.

I expect it's good practice to leave shell scripts as "shell/caller agnostic", so they can be called from root system (Docker containers) or virtual environments: `pipenv run run.sh`, for example.

File: README.md (hunk 1 above)
Comment (andrewmil) [reply README.md]: yep :+1:

**Review by jcox-dev — APPROVED** [review]

**Review by gemmai95 — APPROVED** [review]

**Pull request merged**